src/test/ui/async-await/async-unsafe-fn-call-in-safe.rs

```diff
@@ -0,0 +1,16 @@
+// edition:2018
+
+#![feature(async_await, async_closure)]
+
+struct S;
+
+impl S {
+    async unsafe fn f() {}
+}
+
+async unsafe fn f() {}
```

Comment:
Could you please also extend this to account for calling `async unsafe fn` and `unsafe fn` in `async fn`?

Reply:
done in 60f480dacbc4ee0dbe5bd7dfa6efe0458ea6aab2